src/io/files.js

```diff
@@ -1085,6 +1088,68 @@ p5.prototype.httpDo = function() {
       }
     }
 
+    if (method === 'GET') {
+      // Get the size of the file to be fetched using HEAD request
+      return new Promise(function(resolve, reject) {
+        var xhr = new XMLHttpRequest();
```

Comment:
it looks like there are now two HTTP requests being made per httpDo call.. this one here and the request() on line 1153. is this needed? or is there a way to combine?

Reply:
and another one in getFile() called from line 1116?
I'm wondering if this function just needs to be cleaned up and streamlined so it's clear what's going on.
some inline comments would help a lot too.

Reply:
If a file has to be loaded, and a GET request has to be made, first a HEAD request is made. The HEAD request only fetches the headers and doesn't fetch the entire file so it doesn't take much longer at all. It gives the file size which we use to produce a warning. A normal GET request is then made after the HEAD request is complete to fetch the file the user wants.

So for GET requests, 2 HTTP requests are made, the one in line 1094 and 1116. For other types of requests, just the one in 1116 is made.